### ISSUE TITLE
Propose new cop to detect keyword method names

### ISSUE THREAD

**Issue by viralpraxis** [rubocop/rubocop#13621]
## Is your feature request related to a problem? Please describe.

I've recently stumbled upon this code (minimized):

```ruby
# frozen_string_literal: true

def
def foo
  1
end
```

this is completely valid ruby code which defines `def` method. Obviously the one who wrote it just forgot to remove dangling `def` keyword at line 3, but it led to unexpected consequences.

Running rubocop on this snippet with default configuration leads to 2 offenses:

1. `Style/MethodDefParentheses`, which is sometimes disabled/ignored since some rubyists love parantheses-less arguments list;
2. `Lint/UnusedMethodArgument` which actually might point out the issue but it won't help if e.g. the method name starts with an underscore:

   ```ruby
   # frozen_string_literal: true

   def
   def _foo
     1
   end
   ```

## Describe the solution you'd like

I'd like to propose either

a) a new `Lint/MethodKeywordName` which will check `def` nodes with method names which are [ruby keywords.](https://docs.ruby-lang.org/en/master/syntax/keywords_rdoc.html). There are, as far as I can see, two keywords which are regularry used as method names: `for` and `then`. They can be excluded using `AllowedIdentifiers` configuration option default value.

b) a more specialized cop that only checks for this kind of "double def" code.

## Describe alternatives you've considered

N/A

## Additional context

Here's a minimal working implementation, in case someone wants to check how it aligns with real codebases: https://github.com/rubocop/rubocop/pull/13620

If option a or b is approved I'll finish this implementation

**Comment by dvandersluis** [comment]
A lot of the ruby keywords are also common method names, many of which are used by Ruby itself as method names, so I think this will cause a lot of noise.

Checking for `def def...` is likely the only/most valuable check here due to possible confusion, but even that is very context specific (ie. most cases will result in syntax errors, you need to not use parens and also have no parameters for the "expected" method definition). There are also going to be some cases where users intentionally called a method `def`.

**Comment by koic** [comment]
I fully agree with @dvandersluis's opinion. I have doubts about providing this as a default cop for a general rule.

**Comment by viralpraxis** [comment]
Let me reframe the issue. Is there an existing cop which is configurable so I can disallow specific method names in my codebases? I'm pretty sure if someone from my team ends up with `def def` construct that's a mistake; there are more method identifiers, including `false` and `true`, that I'd like to be disallowed.

**Comment by tejasbubane** [comment]
Would it make sense to add this as configuration option `DisallowedMethodNames` for `Naming/MethodName` cop?
